### ISSUE TITLE
build.rs fails in dependencies for examples

### ISSUE THREAD

**Issue by johalun** [async-rs/async-std#127]
Hey! 
Thanks for this awesome initiative :) 

I can build and run separate binaries that links to async-std as extern crate (like the examples in the readme) but I can't run the examples from within async-std.

```
johannes@jm:~/dev/async-test % rustc -V
rustc 1.39.0-nightly (53df91a9b 2019-08-27)
johannes@jm:~/dev/async-test % cargo -V
cargo 1.39.0-nightly (3f700ec43 2019-08-19)
johannes@jm:~/dev/async-test % uname -a
FreeBSD jm 13.0-CURRENT FreeBSD 13.0-CURRENT r349834+a82ad980c917(dell-fix_iichid-evdev) DELL-NODEBUG  amd64
```

This is what I get
```
johannes@jm:~/dev/async-std % cargo run --example hello-world
   Compiling libnghttp2-sys v0.1.2
   Compiling openssl-sys v0.9.49
   Compiling backtrace-sys v0.1.31
   Compiling mime_guess v2.0.1
   Compiling mime v0.3.13
   Compiling tempdir v0.3.7
   Compiling proc-macro-hack v0.5.9
   Compiling rand_chacha v0.2.1
error: failed to run custom build command for `libnghttp2-sys v0.1.2`

Caused by:
  process didn't exit successfully: `/usr/home/johannes/dev/async-std/target/debug/build/libnghttp2-sys-4c6f3caedee97f80/build-script-build` (signal: 11, SIGSEGV: invalid memory reference)
warning: build failed, waiting for other jobs to finish...
error: failed to run custom build command for `backtrace-sys v0.1.31`

Caused by:
  process didn't exit successfully: `/usr/home/johannes/dev/async-std/target/debug/build/backtrace-sys-78dbde0feafa0d65/build-script-build` (signal: 11, SIGSEGV: invalid memory reference)
--- stdout
cargo:rustc-cfg=rbt

warning: build failed, waiting for other jobs to finish...
error: failed to run custom build command for `openssl-sys v0.9.49`

Caused by:
  process didn't exit successfully: `/usr/home/johannes/dev/async-std/target/debug/build/openssl-sys-7d3ff8c9464a6e09/build-script-main` (signal: 11, SIGSEGV: invalid memory reference)
warning: build failed, waiting for other jobs to finish...
error: build failed
```

Any idea what this might depend on?

**Comment by yoshuawuyts** [comment]
This seems like a problem building nghttp2, which is a dependency of libcurl, which is a dependency of (among other things) Surf. We recently had some linking issues there, and fixed them in 1.0.2 (moved from static builds to dynamic builds).

If you're using Surf (or Isahc) anywhere in your tree, perhaps try updating it?

**Comment by johalun** [comment]
I think this is the issue https://github.com/rust-lang/libc/issues/1489
Closing this.
